apollo-router/src/plugins/subscription.rs

```diff
@@ -181,6 +212,43 @@ pub(crate) struct WebSocketConfiguration {
     pub(crate) protocol: WebSocketProtocol,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Default, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields, default)]
+pub(crate) struct SseConfiguration {
+    /// Path use to get the SSE stream
+    pub(crate) path: Option<String>,
+
+    /// If `reconnect` is `true` (the [default]), the client will automatically
+    /// try to reconnect if the stream ends due to an error. If it is `false`,
+    /// the client will stop receiving events after an error.
+    pub(crate) reconnect: Option<bool>,
+
+    /// If `true` (the [default]) the client will automatically retry the connection, with the
+    /// same delay and backoff behaviour as for reconnects due to stream error.
+    /// If `false`, the client will not retry the initial
+    /// connection.
+    pub(crate) retry_initial: Option<bool>,
+
+    /// After an error, the client will wait this long before the first attempt
+    /// to reconnect.  Subsequent reconnect attempts may wait longer, depending
+    /// on the [`backoff_factor`].
+    #[serde(with = "humantime_serde")]
+    #[schemars(with = "Option<String>")]
+    pub(crate) delay: Option<std::time::Duration>,
+
+    /// Configure the factor by which delays between reconnect attempts will
+    /// exponentially increase, up to [`delay_max`]. The [default] factor is 2,
+    /// so each reconnect attempt will wait twice as long as the previous one.
+    pub(crate) backoff_factor: Option<u32>,
+
+    /// Configure the maximum delay between reconnects (the [default] is 1
+    /// minute). The exponential backoff configured by [`backoff_factor`] will
+    /// not cause a delay greater than this value.
+    #[serde(with = "humantime_serde")]
+    #[schemars(with = "Option<String>")]
+    pub(crate) delay_max: Option<std::time::Duration>,
+}
```

Comment:
As everything is optional, maybe we should add `enabled: bool`  which defaults to `true`. Because in case we just want to enable it with default values it will be an empty object and in yaml it's pretty unusual. What do you think ?

Reply:
It's a bit weird to me having `enabled = false` I made the change and when is not enabled will use the defaults regardless of the values

Reply:
having a `enabled: false` is also a good way to quickly disable it without deleting all the configuration. But yes I understand your concern. It's the pattern we adopted right now when we are in that kind of situation